contracts/ResultComputation.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title ResultComputation
+ * @dev Computes the results of voting on a data value. The voted values, along with their weights, are kept in a sorted
+ * linked list as votes come in. The result is the weighted median of the values, computed by iterating over the list
+ * when needed.
+ * TODO(ptare): All methods are internal because we don't have a migration system written yet, so libraries with public
+ * methods can't yet be linked to from unit tests. 
+ */
+library ResultComputation {
+    using SafeMath for uint;
+
+    uint private constant NULL_NODE_ID = 0;
+
+    struct LinkedListNode {
+        // The total number of tokens that voted for this value.
+        uint weight;
+        // The value this node encapsulates.
+        int value;
+        // Pointer to the next node.
+        uint next;
+    }
+
+    // All the data that needs to be stored to compute results.
+    struct ResultComputationData {
+        // Must be initialized to NULL_NODE_ID. Don't need to do any initialization as long as NULL_NODE_ID == 0.
+        uint head;
+        // The linked list is stored in a mapping from (node #) => node. Each node's next pointer contains the (node #)
+        // of the next node.
+        mapping(uint => LinkedListNode) linkedList;
+
+        // The total weight (total number of tokens) that have voted so far.
+        uint totalWeight;
+        // Number of nodes in the list. Might be able to do something cheeky and use totalWeight instead.
+        uint numberNodes;
+    }
+
+    // Calculates the average of two numbers. Since these are integers, averages of an even and odd number cannot be
+    // represented, and will be rounded down. Signed integer version of OpenZeppelin's Math.average.
+    function average(int a, int b) internal pure returns (int) {
+        // (a + b) / 2 can overflow, so we distribute.
+        return (a / 2) + (b / 2) + ((a % 2 + b % 2) / 2);
+    }
+
+    /**
+     * @dev Iterates through the linked list in ResultComputationData and returns the median.
+     */
+    function getResolvedPrice(ResultComputationData storage data) internal view returns (int) {
+        // TODO(ptare): Figure out what API is used to communicate invalid results.
+        require(data.totalWeight > 0, "Invalid resolved price: no votes were added");
```

Comment:
I think this require is fine - my suggestion would be to add a separate method (in a later PR) called `isComputationValid(data)` or `isVoteValid(data)` that the calling contract can check before it calls this method.